autogpt_platform/frontend/src/components/__legacy__/Wallet.tsx

```diff
@@ -196,16 +197,12 @@ export default function Wallet() {
 
   // Load last seen credits from localStorage once on mount
   useEffect(() => {
-    try {
-      const stored = localStorage.getItem("wallet_last_seen_credits");
-      if (stored !== null) {
-        const parsed = parseFloat(stored);
-        if (!Number.isNaN(parsed)) setLastSeenCredits(parsed);
-        else setLastSeenCredits(0);
-      } else {
-        setLastSeenCredits(0);
-      }
-    } catch {
+    const stored = storage.get(StorageKey.WALLET_LAST_SEEN_CREDITS);
+    if (stored !== undefined && stored !== null) {
+      const parsed = parseFloat(stored);
+      if (!Number.isNaN(parsed)) setLastSeenCredits(parsed);
```

Comment:
can  this be null or not a number?

Reply:
Just a small defensive check. Problem is that `parseFloat(null) => NaN` is of type `number` in Javascript's ( executed at runtime ) dynamic type system 😢 

<img width="243" height="102" alt="Screenshot 2025-10-02 at 19 18 20" src="https://github.com/user-attachments/assets/8dfa720d-46d3-4504-8472-5436db5fb584" />

```ts
 const parsed = parseFloat(stored);
 ```
 Could be `NaN` if stored comes as `""` for example from the browser's local storage, which is unlikely but still.